A Hodge-Podge of Fixes and Features

CodaFi opened the pull request:
- Renames `threadLocalMode` to `threadLocalModel` - LLVM is wrong, not us.
- Adds an accessor to `Switch` instructions to get the default block
- Add atomic ordering parameters to memory access instructions
- Add volatility parameters to memory access instructions
- Add constant inline assembly references.  Note that this is different from module-level inline assembly and should only be done when constructing IR inside a function.  

(Support for module-level ASM is tricky without a better C API)

CodaFi committed:
Use the right name for the Thread Local Model

CodaFi committed:
Add ordering and volatility to memory access insts

CodaFi committed:
Add a getter for the default of a Switch

CodaFi committed:
Add inline assembly

harlanhaskins commented on Sources/LLVM/IRBuilder.swift:
```diff
@@ -1706,6 +1721,48 @@ public class IRBuilder {
+  ///
+  /// - returns: A representation of the newly created inline assembly
+  ///   expression.
+  public func buildInlineAssembly(_ asm: String, type: FunctionType, constraints: String = "", hasSideEffects: Bool = true, needsAlignedStack: Bool = true) -> IRValue {
```
Is there a URL we can point to for more information? Or is this documentation already better than the existing LLVM doc...?

CodaFi replied:
This kind of *is* LLVM's docs on this.  Paraphrased and updated a bit with references to external stuff.

harlanhaskins commented:
LGTM!

CodaFi committed:
Fix atomic ordering accessor

Use the derived ==.  Providing our own causes a
cyclic dependency on the ordering -> LLVM map
which causes crashes on recursive initial accesses
to the dictionary.

segiddins reviewed (approved):
Cool!

CodaFi commented:
⛵️
